### PR TITLE
Cluster aggregate adjustments

### PR DESCRIPTION
**Pull request by jastBytes**
The cluster aggregate is changed in a way that displayname is dropped completley and the name is updateable instead.

### CODE REVIEW

File: pkg/domain/aggregates/cluster.go [hunk 1]
```diff
@@ -142,26 +130,46 @@ func (a *ClusterAggregate) ApplyEvent(event es.Event) error {
...
+		a.apiServerAddr = clusterCreatedV3.GetApiServerAddress()
+		a.caCertBundle = clusterCreatedV3.GetCaCertificateBundle()
 	case events.ClusterUpdated:
 		data := new(eventdata.ClusterUpdated)
 		err := event.Data().ToProto(data)
 		if err != nil {
 			return err
 		}
 
-		if len(data.GetDisplayName()) > 0 && a.displayName != data.GetDisplayName() {
```
Comment (HaniAlshikh) [inline pkg/domain/aggregates/cluster.go:156]: why remove it? `ClusterUpdated(V1)` still has the display name attribute 🤔

File: pkg/domain/formatters/events/cluster.go [hunk 2]
```diff
@@ -97,14 +105,39 @@ func (f *clusterEventFormatter) getFormattedDetailsClusterUpdated(ctx context.Co
 
 	var details strings.Builder
 	details.WriteString(fConsts.ClusterUpdatedDetailsFormat.Sprint(event.Metadata[auth.HeaderAuthEmail]))
-	f.AppendUpdate("Display name", eventData.DisplayName, cluster.DisplayName, &details)
+	f.AppendUpdate("Display name", eventData.DisplayName, cluster.Name, &details)
```
Comment (HaniAlshikh) [inline pkg/domain/formatters/events/cluster.go:108]: for `ClusterUpdated(V1)` display name is still valid, why change it?

**Review by HaniAlshikh — CHANGES_REQUESTED** [review]

File: pkg/domain/aggregates/cluster.go (hunk 1 above)
Comment (jastBytes) [reply pkg/domain/aggregates/cluster.go:156]: Cause it has been removed from the projection as well. So it doesn't make sense to process it anymore for old commands.

File: pkg/domain/formatters/events/cluster.go [hunk 3]
```diff
@@ -97,14 +105,39 @@ func (f *clusterEventFormatter) getFormattedDetailsClusterUpdated(ctx context.Co
...
+		MaxTimestamp: timestamppb.New(event.GetTimestamp().AsTime().Add(time.Duration(-1) * time.Microsecond)), // exclude the update event
+		AggregateId:  &wrapperspb.StringValue{Value: event.AggregateId}},
+	)
+	if err != nil {
+		return "", err
+	}
+
+	var details strings.Builder
+	details.WriteString(fConsts.ClusterUpdatedDetailsFormat.Sprint(event.Metadata[auth.HeaderAuthEmail]))
+	if eventData.Name != nil {
```
[the remark this answers is not in the document]
Comment (jastBytes) [reply pkg/domain/formatters/events/cluster.go:129]: Hm, yeah possible.

[the remark this answers is not in the document]
Comment (jastBytes) [reply pkg/domain/formatters/events/cluster.go:129]: Yeah

File: pkg/domain/aggregates/cluster_test.go [hunk 4]
```diff
@@ -56,7 +56,7 @@ var _ = Describe("Unit Test for Cluster Aggregate", func() {
 		agg := NewClusterAggregate(NewTestAggregateManager())
 
 		ed := es.ToEventDataFromProto(&eventdata.ClusterCreatedV2{
-			DisplayName:         expectedClusterDisplayName,
+			DisplayName:         expectedClusterName,
```
[the remark this answers is not in the document]
Comment (jastBytes) [reply pkg/domain/aggregates/cluster_test.go:59]: That's a bug. Will change. Good find!

**Review by HaniAlshikh — APPROVED** [review]

**Pull request merged**